### DISCUSSION TITLE
Adding metadata to user: Apps are not allowed to perform this mutation

### DISCUSSION THREAD

**Discussion by maxkraanen** [saleor/saleor#15457] [General]
I'm trying to update the metadata of a customer using the API. Somehow the error: "Apps are not allowed to perform this mutation" is thrown which I don't understand because I am allowed to mutate order metadata and I think I should have all permissions. Does anyone have an idea? Thanks!

**Comment by timuric** [comment]
Could you check if you have user permissions?

**Comment by JannikZed** [comment]
weird: suddenly, I'm facing this issue as well. Just for metadata of user - is the user you are trying to update also staff user by any chance? @maxkraanen 
My app definitely has Manage user and manage staff permissions, but still I can't mutate metadata of a user.

It does work to update the metadata of a regular user.

So I guess the issue is: can't update metadata of user that is also staff as app, although all permissions are set.
